Open-Source-Soldier-of-Fortune/changchuanhong-submission/KrustalMST.py

```
@@ -0,0 +1,96 @@
+                self.DepthFirstSearch(i, visited)
+        return visited
+
+    def buildMST(self):
+        """
+        Krustal's Minimum Spanning Tree as described by 
+            Gautier Marti, Frank Nielsen, Mikołaj Bińkowski, Philippe Donnat, 2017,
+            "A review of two decades of correlations, hierarchies, networks and clustering in financial markets" 
+        
+        The function returns an adjacency matrix.
+        """
+        # Reset our previous calculations (Set all edges to null)
+        self.setProcessedDistanceMatrix()
+        # The number of elements in a triangular matrix excluding the diagonal is N(N-1)/2
+        for i in range(int((self.node_count * (self.node_count - 1)) / 2)):
+            # Add edges in order of increasing distances
+            node_a, node_b = self.getIndexOfMinimumDist()
+            # Check if the nodes are already connected by a path; if not, add edges to them
+            visited = self.DepthFirstSearch(node_a, [0]*self.node_count)
+            if not visited[node_b] == 1:
+                self.adjacency_matrix[node_a][node_b] = 1
+                self.adjacency_matrix[node_b][node_a] = 1
+            # Mark the recently selected node to make sure that we will not select it again.
+            self.processed_distance_matrix[node_a][node_b] = self.original_distance_matrix.max() + 1
+        
+        return self.adjacency_matrix
+
+    def getIndexOfMinimumDist(self):
+        """
+        This function gets the index of the matrix elements with the minimum distance from each other.
+        """
+        x = list( (np.where(self.processed_distance_matrix == self.processed_distance_matrix.min())) )
+        return x[0][0], x[1][0]
+
+    def setDistanceMatrix(self, distance_matrix):
+        """
+        :param distance_matrix: A NxN matrix of distances
+
+        This function converts the distance matrix into a numpy array, and checks if the matrix is symmetrical.
+        """
+        try:
```

I am not a fan of using try-catch outside of things like calls to a DB where we can get 3rd party exceptions. This can lead to hiding errors and making debugging very difficult. 

Also, note how you don't expressly list the exception to be caught. This try-catch is a problem waiting to happen.